pkg/gui/presentation/commits.go

```diff
@@ -99,6 +101,14 @@ func GetCommitListDisplayStrings(
 		getGraphLine = func(idx int) string { return "" }
 	}
 
+	branchHeadsToVisualize := lo.FilterMap(branches,
```

Comment:
I'd make this a set for fast access (especially given how many commits we currently iterate through here)

Reply:
See 12f41517e8d2.